app/server/appsmith-plugins/dynamoPlugin/src/main/java/com/external/plugins/DynamoPlugin.java

```diff
@@ -71,6 +91,148 @@ public DynamoPlugin(PluginWrapper wrapper) {
+
+                                return extractedValueMap;
+                            }
+
+                            break;
+                        default:
+                            extractedValueMap.put(entry.getKey(), extractValue(entry.getValue()));
+                    }
+                }
+
+                return extractedValueMap;
+            } else if (rawItem instanceof List) {
+                List<Object> extractedValueList = new ArrayList<>();
+                List<Object> rawItemAsList = (List<Object>) rawItem;
+                for (Object listItem : rawItemAsList) {
+                    extractedValueList.add(extractValue(listItem));
+                }
+
+                return extractedValueList;
+            }
+
+            return rawItem;
+        }
+        /*
+         * - Transform response for easy consumption. For details please visit
+         *   https://github.com/appsmithorg/appsmith/issues/3010
+         */
+        public Object getTransformedResponse(Map<String, Object> rawResponse,
+                                             String action) throws AppsmithPluginException {
+            Map<String, Object> transformedResponse = new HashMap<>();
+
+            if (action.equals(SCAN_ACTION_VALUE)
+                    || action.equals(GET_ITEM_ACTION_VALUE)
+                    || action.equals(PUT_ITEM_ACTION_VALUE)
+                    || action.equals(UPDATE_ITEM_ACTION_VALUE)
+                    || action.equals(DELETE_ITEM_ACTION_VALUE)) {
+                transformedResponse.put(RAW_RESPONSE_LABEL, rawResponse);
+
+                String topLevelKey;
+                switch (action) {
+                    case SCAN_ACTION_VALUE:
+                        topLevelKey = ITEMS_KEY;
+                        break;
+                    case GET_ITEM_ACTION_VALUE:
+                        topLevelKey = ITEM_KEY;
+                        break;
+                    case PUT_ITEM_ACTION_VALUE:
+                    case UPDATE_ITEM_ACTION_VALUE:
+                    case DELETE_ITEM_ACTION_VALUE:
+                        topLevelKey = ATTRIBUTES_KEY;
+                        break;
+                    default:
+                        throw new AppsmithPluginException(
+                                AppsmithPluginError.PLUGIN_ERROR,
+                                "Appsmith has encountered an unexpected error when transforming raw DynamoDb response" +
+                                        ". Please reach out to Appsmith customer support to resolve this."
+                        );
+                }
+
+                for (Map.Entry<String, Object> responseEntry: rawResponse.entrySet()) {
+                    if (!responseEntry.getKey().equals(topLevelKey)) {
+                        transformedResponse.put(responseEntry.getKey(), responseEntry.getValue());
+                    }
+                    else {
+                        if (action.equals(SCAN_ACTION_VALUE)) {
```

Comment:
Minor: As a best practice, always have the constant variable first. This ensures that we don't run the risk of NPEs across the code base.
```suggestion
                        if (SCAN_ACTION_VALUE.equals(action)) {
```

Reply:
Done.